desc/compute/_curve.py

```diff
@@ -646,6 +646,83 @@ def _x_sss_FourierXYZCurve(params, transforms, profiles, data, **kwargs):
     return data
 
 
+def _splinexyz_helper(f, transforms, s_query_pts, method, derivative):
+    """Used to compute XYZ coordinates for the SplineXYZCurve compute functions.
+
+    Parameters
+    ----------
+    f : list of ndarray
+        X, Y, Z function coords with shape (3, len(transforms["knots"]))
+    transforms : dict
+        the transforms from the compute function
+    s_query_pts : ndarray
+        query points that come from s parameterization
+    kwargs : dict
+        the kwargs from the compute function
+    derivative : int
+        derivative order used for interpolation
+
+    Returns
+    -------
+    coords : ndarray
+        Interpolated XYZ coords with shape (3, len(s_query_pts))
+    """
+    f = jnp.asarray(f)
+    transforms["intervals"] = jnp.asarray(transforms["intervals"])
+    has_break_points = len(transforms["intervals"][0])
+
+    def inner_body(f, knots, period=None):
+        """Interpolation for spline curves."""
+        fq = interp1d(
+            s_query_pts,
+            knots,
+            f.T,
+            method=method,
+            derivative=derivative,
+            period=period,
+        )
+
+        return fq.T
+
+    def body(interval, full_f, full_knots):
+        """Body used if there are break points."""
+        istart, istop = interval
+        # catch end-point
+        istop = jnp.where(istop == 0, -1, istop)
+
+        # fill f values outside of interval with break point values so that
+        # interpolation only takes into consideration the interval
```

Comment:
Should note somewhere (maybe in a docstring) that the BC for a given spline is not enforced at the end of each interval but rather at the end of the 0 -> 2pi interval in `s`